### Урок 1. Случайные события. Условная вероятность. Формула Байеса. Независимые испытания

##### 1. Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется ХОТЯ БЫ один туз.

a) В колоде имеется $\frac{52}{4} = 13$ крестовых карт. Вероятность того что первая карта будет крести $\frac{13}{52}$, вероятность что вторая карта будет крести – $\frac{12}{51}$ и так далее

Вероятность того что все черыре карты будут крести$$ \frac{13}{52} \cdot \frac{12}{51} \cdot \frac{11}{50}\cdot \frac{10}{49} = 0.26\%$$

In [6]:
13/52*12/51*11/50*10/49*100

0.26410564225690275

б) возможны 4 случая: 1 туз и 3 не туза, 2 туза и 2 не туза, 3 туза и 1 не туз, 4 туза. Всего в колоде 4 туза и 48 не тузов.

$$ (\frac{4}{52} \cdot \frac{48}{51} \cdot \frac{47}{50}\cdot \frac{46}{49}) +
   (\frac{4}{52} \cdot \frac{3}{51} \cdot \frac{48}{50}\cdot \frac{47}{49}) +
   (\frac{4}{52} \cdot \frac{3}{51} \cdot \frac{2}{50}\cdot \frac{48}{49}) +
   (\frac{4}{52} \cdot \frac{3}{51} \cdot \frac{2}{50}\cdot \frac{1}{49}) = 6.82\%$$

In [7]:
((4/52*48/51*47/50*46/49)+(4/52*3/51*48/50*47/49)+(4/52*3/51*2/50*48/49)+(4/52*3/51*2/50*1/49))*100

6.8235294117647065

##### 2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

найдем количество возможных комбинаций: 

In [8]:
from math import factorial

def combinations (n, k):
    return int(factorial(n) / (factorial(k) * factorial(n -k)))

In [9]:
n = combinations(10, 3)
print(n)

120


вероятность не зная код открыть с первой попытки: $$ \frac{1}{120} = 0.83\%$$

In [12]:
(1/n)*100

0.8333333333333334

##### 3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

$$ \frac{9}{15} \cdot \frac{8}{14} \cdot \frac{7}{13} = 18.46\%$$

In [13]:
(9/15*8/14*7/13)*100

18.46153846153846

##### 4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

$$ \frac{2}{100} \cdot \frac{1}{99} = 0.02\%$$

In [14]:
(2/9900)*100

0.0202020202020202

решение через сочетания

общее число благоприятных сочетаний:

In [15]:
n = combinations(100, 2)
print(n)

4950


число сочетаний два из двух:

In [20]:
m2 = combinations(2, 2)
print(m2)

1


число сочетаний 0 из 98:

In [21]:
m0 = combinations(98, 0)
print(m0)

1


Вероятность того, что из 2 билета будет ровно 2 выигрышных, равна:

In [22]:
P = ((m2 * m0) / n) * 100
print(P)

0.0202020202020202
